Automattic/wp-calypso
# Checkout: Convert useCountryList to useQuery instead of Redux

## Proposed Changes

This refactors the `useCountryList()` React hook (which fetches the list of countries available for payments and the metadata for those countries) to use `useQuery()` instead of the Redux data store and the data-layer. This is part of the overall effort to remove dependence on Redux, particularly in checkout. This hook is also used by purchase management.

> [!NOTE]
> There are other country lists in calypso (`domains`, `sms`, and `woocommerce`) which are not changed by this PR and still use Redux. This only replaces the `payments` country list.

## Testing Instructions

- Visit checkout with a product in your cart. Verify that in the billing info step, the "Country" dropdown is populated.
- Visit `/me/purchases` for a site with at least one subscription. Try to add or change the payment method on the subscription and verify that the "Country" dropdown is populated.
